libs/cardano-ledger-core/testlib/Test/Cardano/Ledger/Plutus/ExUnits.hs

```diff
@@ -0,0 +1,41 @@
+module Test.Cardano.Ledger.Plutus.ExUnits (spec) where
+
+import Cardano.Ledger.Plutus (
+  exBudgetToExUnits,
+  transExUnits,
+ )
+import Test.Cardano.Ledger.Common
+import Test.Cardano.Ledger.Core.Arbitrary ()
+
+spec :: Spec
+spec = do
+  prop "Round-trip to ExBudget" exUnitsToExBudgetRoundTrip
+  prop "Round-trip from ExBudget" exBudgetToExUnitsRoundTrip
+
+-- ExUnits should remain intact when translating to and from the Plutus ExBudget type
+exUnitsToExBudgetRoundTrip :: Gen Property
+exUnitsToExBudgetRoundTrip = do
+  e <- arbitrary
+  let result = exBudgetToExUnits $ transExUnits e
+  pure
+    $ counterexample
+      ( "Before: "
+          <> show e
+          <> "\n After: "
+          <> show result
+      )
+    $ result == Just e
+
+-- Plutus ExBudget should remain intact when translating to and from the ExUnits type
+exBudgetToExUnitsRoundTrip :: Gen Property
+exBudgetToExUnitsRoundTrip = do
+  e <- arbitrary
+  let result = transExUnits <$> exBudgetToExUnits e
+  pure
+    $ counterexample
+      ( "Before: "
+          <> show e
+          <> "\n After: "
+          <> show result
+      )
+    $ result == Just e
```

Comment:
I know you didn't write this, so feel free to keep it the way it is, but just to let you know for the future there is a lot more concise way of writing property tests like this. Using `QuickCheck`s mechanism is very much suboptimal, because not only you are forced to use `counterexample` and  manually print things out, but also you don't get a nice diff when there is a failure, unlike when using `hspec`'s `shouldBe`.
```suggestion
  prop "Round-trip to ExBudget" $ \e -> 
    exBudgetToExUnits (transExUnits e) `shouldBe` Just e
  prop "Round-trip from ExBudget" 
    transExUnits <$> exBudgetToExUnits e `shouldBe` Just e
```